File: web/src/core/translations/professional.ts
import { Messages } from 'lib/intl/messages';
import { ProfessionalType } from 'modules/Professional/ProfessionalType';

export const messages: Messages = {
  PROFESSIONAL: {
    en: 'Professional',
    hr: 'Profesionalac',
  },
  PROFESSIONALS: {
    en: 'Professionals',
    hr: 'Profesionalaci',
  },
  ADD_THERAPIST: {
    en: 'Add therapist',
    hr: 'Dodaj terapeuta',
  },
  THERAPIST: {
    en: 'Therapist',
    hr: 'Terapeut',
  },
  THERAPISTS: {
    en: 'Therapists',
    hr: 'Terapeuti',
  },
  ASSIGN_THERAPIST: {
    en: 'Assign therapist',
    hr: 'Dodijeli terapeuta',
  },
  ASSIGNED_THERAPISTS: {
    en: 'Assigned therapists',
    hr: 'Dodijeljeni terapeuti',
  },
  ASSIGNED_THERAPIST: {
    en: 'Assigned therapist',
    hr: 'Dodijeljeni terapeut',
  },
  ASSIGNED_THERAPISTS_DESCRIPTION: {
    en: 'Only assigned therapists can see this client',
    hr: 'Isključivo dodijeljeni terapeuti imaju pristup ovom klijentu',
  },
  CREATE_PROFESSIONAL: {
    en: 'Create professional',
    hr: 'Kreiraj profesionalca',
  },
  PROFESSIONAL_TITLE: {
    en: 'Title',
    hr: 'Titula',
  },
  PRIMARY_PROFESSIONAL: {
    en: 'Primary',
    hr: 'Primarni',
  },
  [ProfessionalType.Psychotherapist]: {
    en: 'Psychotherapist',
    hr: 'Psihoterapeut',
  },
  [ProfessionalType.Psychologist]: {
    en: 'Psychologist',
    hr: 'Psiholog',
  },
  [ProfessionalType.Psychiatrist]: {
    en: 'Psychiatrist',
    hr: 'Psihijatar',
  },
  [ProfessionalType.Neurologist]: {
    en: 'Neurologist',
    hr: 'Neurolog',
  },
  [ProfessionalType.Pediatrist]: {
    en: 'Pediatrist',
    hr: 'Pedijatar',
  },
  [ProfessionalType.SocialWorker]: {
    en: 'Social worker',
    hr: 'Socijalni radnik',
  },
  [ProfessionalType.Pedagogue]: {
    en: 'Pedagogue',
    hr: 'Pedagog',
  },
  [ProfessionalType.Defectologist]: {
    en: 'Defectologist',
    hr: 'Defektolog',
  },
  [ProfessionalType.Phonetician]: {
    en: 'Phonetician',
    hr: 'Fonetičar',
  },
  [ProfessionalType.Neuropsychiatrist]: {
    en: 'Neuropsychiatrist',
    hr: 'Neuropsihijatar',
  },
  [ProfessionalType.ClinicalPsychologist]: {
    en: 'Clinical psychologist',
    hr: 'Klinički psiholog',
  },
  [ProfessionalType.Supervisor]: {
    en: 'Supervisor',
    hr: 'Supervizor',
  },
  [ProfessionalType.Logoped]: {
    en: 'Logoped',
    hr: 'Logoped',
  },
  [ProfessionalType.Other]: {
    en: 'Other',
    hr: 'Ostalo',
  },
};

File: pkg/professional/professional.go
package professional

import "github.com/domonda/go-types/language"

type Type string

const (
	TypePsychotherapist      Type = "PSYCHOTHERAPIST"
	TypePsychologist         Type = "PSYCHOLOGIST"
	TypePsychiatrist         Type = "PSYCHIATRIST"
	TypeNeurologist          Type = "NEUROLOGIST"
	TypePediatrist           Type = "PEDIATRIST"
	TypeSocialWorker         Type = "SOCIAL_WORKER"
	TypePedagogue            Type = "PEDAGOGUE"
	TypeDefectologist        Type = "DEFECTOLOGIST"
	TypePhonetician          Type = "PHONETICIAN"
	TypeNeuropsychiatrist    Type = "NEUROPSYCHIATRIST"
	TypeClinicalPsychologist Type = "CLINICAL_PSYCHOLOGIST"
	TypeSupervisor           Type = "SUPERVISOR"
	TypeLogoped              Type = "LOGOPED"
	TypeOther                Type = "OTHER"
)

func (t Type) Message(lang language.Code) string {
	switch t {
	case TypePsychotherapist:
		if language.BA == lang {
			return "Psihoterapeut"
		}
		return "Psychotherapist"
	case TypePsychologist:
		if language.BA == lang {
			return "Psiholog"
		}
		return "Psychologist"
	case TypePsychiatrist:
		if language.BA == lang {
			return "Psihijatar"
		}
		return "Psychiatrist"
	case TypeNeurologist:
		if language.BA == lang {
			return "Neurolog"
		}
		return "Neurologist"
	case TypePediatrist:
		if language.BA == lang {
			return "Pediatar"
		}
		return "Pediatrist"
	case TypeSocialWorker:
		if language.BA == lang {
			return "Socialni radnik"
		}
		return "Social worker"
	case TypePedagogue:
		if language.BA == lang {
			return "Pedagog"
		}
		return "Pedagogue"
	case TypeDefectologist:
		if language.BA == lang {
			return "Defektolog"
		}
		return "Defectologist"
	case TypePhonetician:
		if language.BA == lang {
			return "Fonetičar"
		}
		return "Phonetician"
	case TypeNeuropsychiatrist:
		if language.BA == lang {
			return "Neuropsihijatar"
		}
		return "Neuropsychiatrist"
	case TypeClinicalPsychologist:
		if language.BA == lang {
			return "Klinički psiholog"
		}
		return "Clinical psychologist"
	case TypeSupervisor:
		if language.BA == lang {
			return "Supervizor"
		}
		return "Supervisor"
	case TypeLogoped:
		if language.BA == lang {
			return "Logoped"
		}
		return "Logoped"
	case TypeOther:
		if language.BA == lang {
			return "Ostalo"
		}
		return "Other"
	default:
		return ""
	}
}
